### PR TITLE
Avoid caching in Mongo QuerySet

### PR DESCRIPTION
**Pull request by tsx**
Just a simple helper to avoid unintended caching in mongoengine querysets.

### CODE REVIEW

File: tests/__init__.py [hunk 1]
```diff
@@ -798,6 +799,40 @@ class TestEnum(DeclEnum):
...
+        D.drop_collection()
+
+        for i in range(2):
+            D(i=i).save()
+
+        self.assertTrue(is_cached(D.objects.all()))
+        self.assertFalse(is_cached(iter_no_cache(D.objects.all())))
+
+        # check correct exit behavior
+        self.assertEqual([d.i for d in iter_no_cache(D.objects.all())], list(range(2)))
```
Comment (thomasst) [inline tests/__init__.py]: Is the order guaranteed or do we need to do a set comparison?

File: tests/__init__.py [hunk 2]
```diff
@@ -798,6 +799,40 @@ class TestEnum(DeclEnum):
...
+        self.assertEqual([d.i for d in iter_no_cache(D.objects.limit(1))], list(range(1)))
+
```
Comment (thomasst) [inline tests/__init__.py]: We should add a unit test where we test iterating over the queryset with number_of_objects > batch_size.

File: flask_common/documents.py [hunk 3]
```diff
@@ -438,3 +438,18 @@ def _get_query_shape(self, query):
...
+def iter_no_cache(query_set):
+    """Iterate over queryset without caching it.
+
+    Useful for iterating over large result sets / bulk actions.
+
+    If a batch size is not set, apply a sensible default of 1000
+    that's better than what Mongo server is doing (101 first and
+    then as many as it can fit in 4MB) to avoid cursor timeouts.
+    """
+    if query_set._batch_size is None:
```
Comment (tsx) [inline flask_common/documents.py]: @thomasst I also added a default batch size in case anyone forgets an explicit one.

File: tests/__init__.py [hunk 4]
```diff
@@ -798,6 +799,41 @@ class TestEnum(DeclEnum):
...
+        for i in range(10):
+            D(i=i).save()
+
+        self.assertTrue(is_cached(D.objects.all()))
+        self.assertFalse(is_cached(iter_no_cache(D.objects.all())))
+
+        # check for correct exit behavior
+        self.assertEqual({d.i for d in iter_no_cache(D.objects.all())}, set(range(10)))
+        self.assertEqual({d.i for d in iter_no_cache(D.objects.all().batch_size(5))}, set(range(10)))
+        self.assertEqual({d.i for d in iter_no_cache(D.objects.limit(1))}, set(range(1)))
```
Comment (thomasst) [inline tests/__init__.py]: This still assumes order, so you'd either have to add an order clause, or assume that `d.i` may be an arbitrary number.

**Pull request merged**